### PR TITLE
fix: withdrawn & canceled tab added in speaker page

### LINKED ISSUE #5871: Add 'Withdrawn' and 'canceled' tab in view Speakers page.
**Describe the bug**

Currently 'withdrawn' and 'canceled' tab is not present in view speakers page. 

**To Reproduce**
Steps to reproduce the behaviour:

Create an event first.
1. Go to 'manage events'
2. Click on 'event dashboard'
3. Got to speakers tab.

**Expected behaviour**

Add 'Withdrawn' and 'canceled' tab in view Speakers page.

**Screenshots**

now

**Additional context**

you can refer view session page. It already have withdarwn and canceled tabs.

expected

### PR DESCRIPTION
**Pull request by MeghalBisht**
Fixes #5871 

#### Short description of what this resolves:
Added 'Withdrawn' and 'canceled' tab in view Speakers page.

**Screenshot** - 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by daretobedifferent18 — COMMENTED** [review]
Everything is working fine but I can't see any difference between withdraw and cancelled .
speakers, bcz both withdrawn and cancelled speakers are comming under both tabs

Screenshot_2020-12-05 Speakers test Events eventyay(2) 
Screenshot_2020-12-05 Speakers test Events eventyay(1)

**Comment by MeghalBisht** [comment]
> Everything is working fine but I can't see any difference between withdraw and cancelled .
> speakers, bcz both withdrawn and cancelled speakers are comming under both tabs

@daretobedifferent18 Thanks, its updated now. Can you check once?

File: app/routes/events/view/speakers/list.js [hunk 1]
```diff
@@ -76,6 +82,30 @@ export default class extends Route.extend(EmberTableRouteMixin) {
...
+          op   : 'any',
+          val  : {
+            name : 'state',
+            op   : 'eq',
+            val  : 'withdrawn'
+          }
+        }
+      ];
+    } else if (params.speakers_status === 'canceled') {
+      filterOptions = [
```
Comment (maze-runnar) [inline app/routes/events/view/speakers/list.js:98]: I think these conditions can be replaced by  - 
```
if (params.speakers_status) {
      filterOptions = [
           {
          name : 'sessions',
          op   : 'any',
          val  : {
            name : 'state',
            op   : 'eq',
            val  : params.speakers_status
          }
        }
      ];
    }
```

Comment (MeghalBisht) [reply app/routes/events/view/speakers/list.js:98]: what about "all" ?

Comment (maze-runnar) [reply app/routes/events/view/speakers/list.js]: take that in else

Comment (MeghalBisht) [reply app/routes/events/view/speakers/list.js]: Check now please, thanks

**Comment by MeghalBisht** [comment]
**Screenshots**

File: app/routes/events/view/speakers/list.js [hunk 2]
```diff
@@ -10,6 +10,12 @@ export default class extends Route.extend(EmberTableRouteMixin) {
         return this.l10n.t('Accepted');
       case 'rejected':
         return this.l10n.t('Rejected');
+      case 'confirmed':
+        return this.l10n.t('Confirmed');
+      case 'withdrawn':
+        return this.l10n.t('Withdrawn');
+      case 'canceled':
+        return this.l10n.t('Canceled');
```
Comment (maze-runnar) [inline app/routes/events/view/speakers/list.js:18]: Please use this - 
```
import { capitalize } from 'lodash-es';
titleToken() {
    if (['accepted', 'rejected', 'confirmed', 'withdrawn', 'canceled'].includes(this.params.speakers_status)) {
      return capitalize(this.params.invoice_status);
    } else {
      return this.l10n.t('All');
    }
  }

```
for more info - 
https://github.com/fossasia/open-event-frontend/blob/5c50d993703c8aef1a043ae92b4a469e2e43c894/app/routes/account/billing/invoices/list.js

Comment (iamareebjamal) [reply app/routes/events/view/speakers/list.js:18]: Wrap capitalize in tVar

File: app/routes/events/view/speakers/list.js [hunk 3]
```diff
@@ -1,21 +1,13 @@
...
-      case 'rejected':
-        return this.l10n.t('Rejected');
-      case 'confirmed':
-        return this.l10n.t('Confirmed');
-      case 'withdrawn':
-        return this.l10n.t('Withdrawn');
-      case 'canceled':
-        return this.l10n.t('Canceled');
+    if (['accepted', 'rejected', 'confirmed', 'withdrawn', 'canceled'].includes(this.params.speakers_status)) {
+      return capitalize(this.params.invoice_status);
```
Comment (iamareebjamal) [inline app/routes/events/view/speakers/list.js:8]: tVar

Comment (MeghalBisht) [reply app/routes/events/view/speakers/list.js:8]: done

Comment (MeghalBisht) [reply app/routes/events/view/speakers/list.js]: @iamareebjamal  Should I squash my commits?

File: app/routes/events/view/speakers/list.js [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
   titleToken() {
-    switch (this.params.speakers_status) {
-      case 'pending':
-        return this.l10n.t('Pending');
-      case 'accepted':
-        return this.l10n.t('Accepted');
-      case 'rejected':
-        return this.l10n.t('Rejected');
+    if (['accepted', 'rejected', 'confirmed', 'withdrawn', 'canceled'].includes(this.params.speakers_status)) {
+      return this.l10n.tVar(capitalize(this.params.invoice_status));
```
Comment (maze-runnar) [inline app/routes/events/view/speakers/list.js:8]: a last thing, please update `this.params.invoice_status` to `this.params.speakers_status`

File: app/routes/events/view/speakers/list.js [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
-      case 'pending':
-        return this.l10n.t('Pending');
-      case 'accepted':
-        return this.l10n.t('Accepted');
-      case 'rejected':
-        return this.l10n.t('Rejected');
+    if (['accepted', 'rejected', 'confirmed', 'withdrawn', 'canceled'].includes(this.params.speakers_status)) {
+      return this.l10n.tVar(capitalize(this.params.speakers_status));
+    } else {
+      return this.l10n.tVar('All');
```
Comment (iamareebjamal) [inline app/routes/events/view/speakers/list.js:10]: Why is this tVar? This is static, not variable

Comment (MeghalBisht) [reply app/routes/events/view/speakers/list.js]: I dropped that commit by mistake, pls check now

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**